File: src/main/kotlin/ReverseNumber.java
public class ReverseNumber {
    public static void main(String[] args){

        int number=23456;
        int reverseNumber=findReverseNumberBYWhileLoop(number);
        System.out.println(reverseNumber);
    }

    public static int findReverseNumberBYForLoop(int num){
        int reverseNum=0;
        for (;num!=0; num/=10){
            int num1=num%10;
            reverseNum= reverseNum*10+num1;
        }
        return reverseNum;
    }

    public static int findReverseNumberBYWhileLoop(int num){
        int reverseNum=0;
        while (num!=0){
            int digit=num%10;
            reverseNum=reverseNum*10+digit;
            num /=10;
        }
        return reverseNum;
    }
}

File: src/main/programs/generalprograms/ReverseNumber.kt
package generalprograms

fun main() {
    var num = 12345

    println(reverseNumber(num))
    println(reverseNumber1(num))
}

//write program of reverse the number Using while loop
fun reverseNumber(num: Int): Int {
    var num1=num
    var rev = 0
    while (num1 != 0) {
        var rem = num1 % 10
        rev = rev * 10 + rem
        num1 /= 10
    }
    return rev
}

//write program of reverse the number Using for loop
fun reverseNumber1(num: Int): Int{
    var n=num
    var rev=0
    val digits=num.toString().length
    for ( i in 1..digits){
        val rem= n%10
        rev=rev*10+rem
        n/=10
       // if (n==0) break
    }
    return rev
}

